app/models/availability.rb

```diff
@@ -39,4 +41,15 @@ def self.process_data
         { 'Date' => date, 'Count' => count }
       end
   end
+
+  private
+
+  def cancel_duplicates
+    previous_availabilities = person.availabilities.for_time_span(start_time..end_time).order(:created_at)
+    previous_availabilities.each do |a|
+      if start_time == a.start_time && end_time == a.end_time
+        a.update(status: "Cancelled") unless self == a
```

Comment:
Is there a reason you are using `for_time_span(start_time..end_time)` instead of just using an equality search to get precisely the records that need to be cancelled? e.g., something like `person.availabilities.where(start_time: self.start_time, end_time: self.end_time).where.not(id: self.id)`

I think that way you can omit `unless self == a` and the time comparison, since you'll avoid loading the `self` record and anything that doesnt match.